File: py/Number_of_1_Bits.py
class Solution:
    def hammingWeight(self, n: int) -> int:
        count = 0
        while n:
            n &= n - 1
            count += 1
        return count



        ## method 2?
        # nums = str(n)
        # count = 0
        # for i in range(len(nums)):
        #     if nums[i] == '1':
        #         count += 1
        # return count

if __name__ == '__main__':
    s = Solution()
    print(s.hammingWeight(1011))

    # 1101 1100 1000 0000
    # 1100 1011 0111

File: c_plus/Number_of_1_Bits.cpp
#include<iostream>
using namespace std;

class Solution {
public:
    int hammingWeight(uint32_t n) {
        int cnt = 0;
        for (int i = 0; i < 32; i++) {
            if (n & 1) cnt++;
            n /= 2;
        }
        return cnt;

        int cnt = 0;
        for (int i = 0; i < 32; i++) {
            if (n & 1) cnt++;
            n >>= 1;
        }
        return cnt;
    }
};

int main() {
    Solution s;
    cout << s.hammingWeight(11) << endl;
}
